src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -48,8 +48,8 @@ internal class SecureChannel
 
         private bool _refreshCredentialNeeded;
 
-        private readonly Oid _serverAuthOid = new Oid("1.3.6.1.5.5.7.3.1");
-        private readonly Oid _clientAuthOid = new Oid("1.3.6.1.5.5.7.3.2");
+        private readonly Oid _serverAuthOid = new Oid("1.3.6.1.5.5.7.3.1", "Server Authentication");
+        private readonly Oid _clientAuthOid = new Oid("1.3.6.1.5.5.7.3.2", "Client Authentication");
```

Comment:
These OIDs aren't in https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs#L110, which means that Windows localizes the string.  So if the Oid.FriendlyName can ever be exposed you'll be enforcing English.

Passing `null` would be valid for most contexts, as long as it's not being shared out.  (And if it's not being shared out, it could be static readonly...)

```C#
const OidGroup CRYPT_OID_DISABLE_SEARCH_DS_FLAG = (OidGroup)(unchecked((int)0x80000000));
const OidGroup LocalEku = CRYPT_OID_DISABLE_SEARCH_DS_FLAG | OidGroup.EnhancedKeyUsage);
_clientAuthInfo = Oid.FromOidValue("1.3.6.1.5.5.7.3.2", LocalEku);
```

Probably works.